src/components/studentlist.tsx

```diff
@@ -0,0 +1,242 @@
+//import React from 'react';
+import React,{ useEffect } from 'react';
+import ListGroup from 'react-bootstrap/ListGroup';
+import LectureItem from "./lectureItem"
+import CourseItem from "./CourseItem"
+import StudentItem from "./studentItem"
+import axios from "axios";
+import Calendarweek from "./Calendarweek"
+
+
+
+
+  class studentList extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    this.state={lectures:[],students:[],courses:[],selectedLecture:0,selectedCourse:0};
+ 
+  }
+  
+  getStudentList = (lectureID) => {
+     
+    axios  
+      .get(`http://localhost:4001/api/lectures/:${lectureID}/students`, {})
```

Comment:
As I specified in my #2, there is no need to call `localhost:4001`, you could just do 
```js
 axios  
      .get(`/lectures/:${lectureID}/students`, {})...
```

Reply:
You mean Instead of absolute path , I should use relative path?

Reply:
Yes, you should also delete the `/api/` prefix as the frontend has access to the server through a proxy: 
https://github.com/giovannitangredi/PULSeBS/blob/528a71275ca6e739cf4dee31a7f7ddf531aba489/package.json#L28

Reply:
I changed the address as you said but still I get error:
Failed to load resource: the server responded with a status of 401 (Unauthorized)

Reply:
The route is protected by authentication, it will respond with `401` if you make the request without logging in first.

Reply:
So, are we going to wait for the login? is there a way to use the routes without login just for testing purposes?

Reply:
So, are we going to wait for the login? is there a way to use the routes without login just for testing purposes?

Reply:
There are two ways: 

1. Use Postman to send a post request to the login route ( http://localhost:3000/auth/login ) with the following body:  `{"email": "s274930@studenti.polito.it", "password": "password"}`; then copy the cookie in the response in your browser. Or
2. move temporarily the route you are interested in above `app.use("/api/auth", ..` so that is not protected by authentication, then temporarily modify the controllers you want to test so that you set in some way the parameters that before were set using JWT.

Reply:
as you said and as I remember from WEBAPP1 I can bring app.use("/api/lectures", lectureRouter); above every thing .
```
app.use(morgan("tiny"));

app.use("/api/user", userRouter);
app.use("/api/courses", courseRouter);
app.use("/api/lectures", lectureRouter);

// Routes
app.use("/api/auth", authRouter);
```
but now it shows error 404 , not found

Reply:
I doubt, login works otherwise you would get `Unauthorized access. ` error.

Reply:
No I brought some routes above login
```// Note: Keep this at the top, above routes
app.use(cors());
app.use(helmet());
app.use(compression());
app.use(bodyParser.urlencoded({ extended: false }));
app.use(bodyParser.json());
app.use(cookieParser());
// Set-up logging
app.use(morgan("tiny"));

app.use("/api/user", userRouter);
app.use("/api/courses", courseRouter);
app.use("/api/lectures", lectureRouter);

// Routes
app.use("/api/auth", authRouter);
// For the rest of the code, all APIs require authentication```